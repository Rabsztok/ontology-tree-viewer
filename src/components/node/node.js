import React from 'react'
import Branch from 'components/branch/branch'
import Leaf from 'components/leaf/leaf'
import { observer } from 'mobx-react'

const Node = observer(({ element: { children, active, toggle, ...props } }) => {
  if (children) {
    return (
      <Branch active={active} {...props}>
        {children.map(child => <Node key={child.id} element={child} />)}
      </Branch>
    )
  } else return <Leaf active={active} toggle={toggle} {...props} />
})

export { Node }
export default Node
