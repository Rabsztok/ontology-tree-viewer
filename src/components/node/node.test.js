import React from 'react'
import { shallow } from 'enzyme'
import { Node } from './node'

describe('Node', () => {
  it('renders branch without errors', () => {
    const element = {
      name: 'branch',
      active: false,
      children: [{ id: 2, name: 'child' }]
    }
    const component = shallow(<Node element={element} />)
    expect(component).toMatchSnapshot()
  })

  it('renders leaf without errors', () => {
    const element = { name: 'branch', active: false, toggle: () => {} }
    const component = shallow(<Node element={element} />)
    expect(component).toMatchSnapshot()
  })
})
