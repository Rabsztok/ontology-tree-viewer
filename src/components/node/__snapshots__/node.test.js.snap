// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node renders branch without errors 1`] = `
<Branch
  active={false}
  name="branch"
>
  <_class
    element={
      Object {
        "id": 2,
        "name": "child",
      }
    }
    key="2"
  />
</Branch>
`;

exports[`Node renders leaf without errors 1`] = `
<withRouter(Leaf)
  active={false}
  name="branch"
  toggle={[Function]}
/>
`;
